packages/peer-store/src/utils/address-to-fqmultiaddr.ts

```diff
@@ -0,0 +1,27 @@
+import { CodeError } from '@libp2p/interface/errors'
+import { isMultiaddr } from '@multiformats/multiaddr'
+import { codes } from '../errors.js'
+import type { PeerId } from '@libp2p/interface/peer-id'
+import type { Address } from '@libp2p/interface/peer-store'
+
+/**
+ * Converts {@link Address}[] for a peer that may not contain the {@link PeerId} to {@link Address}[]
+ * that do contain the {@link PeerId}.
+ */
+export function addressToFqMultiaddr (peerId: PeerId, addresses: Address[]): Address[] {
```

Comment:
A nit, but the output type is `Address[]` not `Multiaddr` and they are already fully qualified (assuming that's what `fq` refers to here) as you don't necessarily need the peer id to dial them so this could probably have a better name like `addressesWithPeerIds` or something?